test/org/sagebionetworks/bridge/validators/IdentifierUpdateValidatorTest.java

```diff
@@ -0,0 +1,80 @@
+package org.sagebionetworks.bridge.validators;
+
+import static org.sagebionetworks.bridge.TestUtils.assertValidatorMessage;
+
+import org.junit.Test;
+import org.sagebionetworks.bridge.TestConstants;
+import org.sagebionetworks.bridge.models.accounts.IdentifierUpdate;
+import org.sagebionetworks.bridge.models.accounts.Phone;
+import org.sagebionetworks.bridge.models.accounts.SignIn;
+
+public class IdentifierUpdateValidatorTest {
+
+    // validates sign by phone or email
+    // validates sign in by reauth
+    
+    @Test
+    public void signInRequired() {
+        IdentifierUpdate update = new IdentifierUpdate(null, "updated@email.com", null);
+        
+        assertValidatorMessage(IdentifierUpdateValidator.INSTANCE, update, "IdentifierUpdate",
+                "requires a signIn object");
+    }
+    
+    @Test
+    public void signInErrorsNested() {
+        // Sign in with no password
+        SignIn signIn = new SignIn.Builder().withStudy(TestConstants.TEST_STUDY_IDENTIFIER)
+                .withEmail(TestConstants.EMAIL).build();
+        
+        IdentifierUpdate update = new IdentifierUpdate(signIn, "updated@email.com", null);
+        assertValidatorMessage(IdentifierUpdateValidator.INSTANCE, update, "signIn.password", "is required");
+    }
```

Comment:
Is it worth having a separate test for nested signIn error vs nested reauth error? This way, we test both branches of the if statement.